File: CPP/BOJ-B1-11655-ROT13.cpp
#include <bits/stdc++.h>
using namespace std;

string s;

int main()
{
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);

  // freopen("input.txt", "r", stdin);
  // freopen("output.txt", "w", stdout);

  getline(cin, s);

  for (int i = 0; i < s.size(); i++)
  {
    // if (s[i] >= 65 && s[i] <= 90)
    // {
    //   if (s[i] + 13 > 90)
    //     s[i] = s[i] + 13 - 26;
    //   else
    //     s[i] = s[i] + 13;
    // }
    // else if (s[i] >= 97 && s[i] <= 122)
    // {
    //   if (s[i] + 13 > 122)
    //     s[i] = s[i] + 13 - 26;
    //   else
    //     s[i] = s[i] + 13;
    // }

    if (s[i] >= 65 && s[i] <= 90)
      s[i] = (s[i] - 65 + 13) % 26 + 65;
    else if (s[i] >= 97 && s[i] <= 122)
      s[i] = (s[i] - 97 + 13) % 26 + 97;

    cout << s[i];
  }

  return 0;
}

File: JS/BOJ-B1-11655-ROT13.js
"use strict";

const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let input = fs
  .readFileSync(filePath)
  .toString()
  .split("");

let output = [];
const solution = (input) => {
  const upper = /[A-Z]/;
  const lower = /[a-z]/;

  for(let i = 0; i < input.length; ++i){
    if(upper.test(input[i])) input[i] = String.fromCharCode((input[i].charCodeAt(0) - 65 + 13) % 26 + 65);
    else if (lower.test(input[i])) input[i] = String.fromCharCode((input[i].charCodeAt(0) - 97 + 13) % 26 + 97);
  }

  output = input;
};

solution(input);
fs.writeFileSync("/dev/stdout", output.join(""));
// fs.writeFileSync("../output.txt", output.join(""), (err) => console.error);
